core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2220,7 +2210,22 @@ class ReplicaManager(val config: KafkaConfig,
     replicaFetcherManager.removeFetcherForPartitions(partitionsToMakeFollower.keySet)
```

Comment:
Perhaps it's worth a comment here that stopping the fetchers is required so that we can initialize the fetch position correctly. It's a subtle point which we might miss again in the future.